code_search/pipeline/index_update_pipeline.py

```diff
@@ -0,0 +1,127 @@
+                           job_name: str, worker_machine_type: str,
+                           num_workers: int, step_name='dataflow_function_embedding'):
+    return default_gcp_op(
+        name = step_name,
+        image = 'gcr.io/yang-codesearch/code-search-dataflow:v20181124-23deb39',
+        command = [
+            'python2',
+            '-m',
+            'code_search.dataflow.cli.create_function_embeddings',
+        ],
+        arguments = [
+            '--project', project,
+            '--runner', runner,
+            '--target_dataset', target_dataset,
+            '--problem', problem,
+            '--data_dir', data_dir,
+            '--saved_model_dir', saved_model_dir,
+            '--job_name', job_name,
+            '--temp_location', temp_location,
+            '--staging_location', staging_location,
+            '--worker_machine_type', worker_machine_type,
+            '--num_workers', num_workers,
+            '--requirements_file', 'requirements.dataflow.txt',
+            '--wait_until_finished',
+        ]
+    )
+
+def search_index_creator_op(working_dir: str, data_dir: str, component: str, cluster_name: str, workflow_id: str):
+    return dsl.ContainerOp(
+        # use component name as step name
+        name = component,
+        image = 'gcr.io/yang-codesearch/code-search-search-index-creator:v20181126-6a63824',
+        arguments = [
+            '--working_dir', working_dir,
+            '--data_dir', data_dir,
+            '--component', component,
+            '--cluster', cluster_name,
+            '--workflow_id', workflow_id,
+        ]
+    )
+
+# The pipeline definition
+@dsl.pipeline(
+    name='function_embedding',
+    description='Example function embedding pipeline'
+)
+def function_embedding_update(
+        project,
+        working_dir,
+        data_dir,
+        saved_model_dir,
+        cluster_name,
+        problem=dsl.PipelineParam(name='problem', value='kf_github_function_docstring'),
+        runner=dsl.PipelineParam(name='runnder', value='DataflowRunner'),
+        target_dataset=dsl.PipelineParam(name='target-dataset', value='code_search'),
+        worker_machine_type=dsl.PipelineParam(name='worker-machine-type', value='n1-highcpu-32'),
+        num_workers=dsl.PipelineParam(name='num-workers', value=5)):
+    workflow_name = '{{workflow.name}}'
+    temp_location = '%s/dataflow/%s/temp' % (working_dir, workflow_name)
+    staging_location = '%s/dataflow/%s/staging' % (working_dir, workflow_name)
+    function_embedding = dataflow_function_embedding_op(project, runner, target_dataset,problem,data_dir,saved_model_dir,
+                                        temp_location,staging_location,workflow_name,worker_machine_type,num_workers)
+    index_creator = search_index_creator_op(working_dir, data_dir, 'search-index-creator', cluster_name,
+                               workflow_name).after(function_embedding)
+
+
+if __name__ == '__main__':
+  import kfp.compiler as compiler
```

Comment:
Why is this imported here and not at the top?

Can you set the logging info like we do in other main functions
https://github.com/kubeflow/examples/blob/master/code_search/src/code_search/dataflow/cli/create_function_embeddings.py#L57

Reply:
In general we should configure onfigure the logging module in main so that code can see logging to output info mesages.